app/controllers/posts_controller.rb

```diff
@@ -7,5 +8,28 @@ def index
   def show
     @post = Post.includes(comments: :author).find(params[:id])
     @user = @post.author
+    @new_like = Like.new
+  end
+
+  def new
+    @user = current_user
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.new(post_params)
+    @user = User.find(params[:user_id])
+
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post successfully created'
+    else
+      redirect_to new_user_post_path(@user), alert: 'Failed to create post'
+    end
+  end
```

Comment:
- [ ] When trying to create a post, I got an error. Kindly fix this error  👇 

![rails form post](https://github.com/ReemMohamedAbdelfatah/rails-blog-app/assets/66097371/593d99da-6638-4ce7-980d-c427ded249f6)